pertpy/tools/_differential_gene_expression/_simple_tests.py

```diff
@@ -127,13 +148,24 @@ def _get_idx(column, value):
 
         res_dfs = []
         baseline_idx = _get_idx(column, baseline)
-        for group_to_compare in groups_to_compare:
-            comparison_idx = _get_idx(column, group_to_compare)
-            res_dfs.append(
-                model._compare_single_group(baseline_idx, comparison_idx, paired=paired, **test_kwargs).assign(
-                    comparison=f"{group_to_compare}_vs_{baseline if baseline is not None else 'rest'}"
-                )
+
+        if permutation_test:
+            test_kwargs = dict(test_kwargs)
+            test_kwargs.update({"test": permutation_test, "n_permutations": n_permutations})
+        elif permutation_test is None and cls.__name__ == "PermutationTest":
+            logger.warning("No permutation test specified. Using WilcoxonTest as default.")
+
+        comparison_indices = [_get_idx(column, group_to_compare) for group_to_compare in groups_to_compare]
```

Comment:
If I get this right, this implements parallelism only at the level of comparisons. This means, if there's only one comparison, there would be no benefit of parallelization. I think it would be more beneficial to implement parallelism at the level of variables, i.e. in `_compare_single_group` at `for var in tqdm(self.adata.var_names):`

Reply:
The permutation_test also has a `vectorize` attribute which allows to pass entire matrices to the test. This likely also speeds up testing quite a bit when whole blocks of data are processed together. Maybe we even get implicit parallelism through the BLAS backend of numpy when doing so.

Reply:
It probably requires a bit of testing to find out what's faster. Per-variable paralleism, just passing the entire array into `stats.permutation_test(..., vectorized=True)` or splitting the data in chunks and then passing it to `permutation_test(..., vectorized=True)`.